File: database/models/outbound.go
package models

import "time"

type Outbound struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	Tag       string    `gorm:"size:64;uniqueIndex;not null" json:"tag"`
	Type      string    `gorm:"size:32;not null" json:"type"` // direct, block, dns, warp, wireguard, socks, http, custom
	Server    string    `gorm:"size:255" json:"server"`
	Port      int       `gorm:"default:0" json:"port"`
	Settings  string    `gorm:"type:text" json:"settings"` // Extra JSON parameters (e.g. WARP private key, peer IP, reserved, etc.)
	Enable    bool      `gorm:"default:true" json:"enable"`
	Remark    string    `gorm:"size:255" json:"remark"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: frontend/src/api/outbound.ts
import request from './request'

export interface Outbound {
  id?: number
  tag: string
  type: string // direct, block, dns, warp, wireguard, socks, http, custom
  server?: string
  port?: number
  settings?: string
  enable: boolean
  remark?: string
  username?: string
  password?: string
  uuid?: string
  flow?: string
  security?: string
  sni?: string
  fingerprint?: string
  alterId?: number
  security_method?: string
  obfs_type?: string
  obfs_password?: string
  up_mbps?: number
  down_mbps?: number
  method?: string
  private_key?: string
  peer_public_key?: string
  pre_shared_key?: string
  reserved?: string
  mtu?: number
  local_address_ipv4?: string
  local_address_ipv6?: string
  detour?: string
  outbounds?: string[]
  url?: string
  interval?: string
}

export const getOutbounds = () => request.get<any, Outbound[]>('/outbounds')
export const createOutbound = (data: Partial<Outbound>) => request.post<any, Outbound>('/outbounds', data)
export const updateOutbound = (id: number, data: Partial<Outbound>) => request.put<any, Outbound>(`/outbounds/${id}`, data)
export const deleteOutbound = (id: number) => request.delete<any, any>(`/outbounds/${id}`)
export const pingOutbound = (id: number) => request.post<any, { latency: number }>(`/outbounds/${id}/ping`)
export const getRawConfig = () => request.get<any, { config: string }>('/server/config')
